src/glslplugin/lang/parser/GLSLParsing.java

```diff
@@ -266,13 +293,25 @@ private boolean parseExternalDeclaration() {
                 b.error("Empty interface block is not allowed.");
             }
 
+            while (b.getTokenType() == PREPROCESSOR_BEGIN) {
+                parsePreprocessor();
+            }
+
             while (!tryMatch(RIGHT_BRACE) && !eof()) {
+
                 final PsiBuilder.Marker member = b.mark();
                 parseQualifierList(true);
                 if (!parseTypeSpecifier()) b.advanceLexer();
                 parseDeclaratorList();
-                match(SEMICOLON, "Expected ';'");
+                if (b.getTokenType() == SEMICOLON) {
+                    b.advanceLexer(false, true); // we need to check for preprocessor ourselves here
+                } else {
+                    b.error("Expected ';'");
+                }
                 member.done(STRUCT_MEMBER_DECLARATION);//TODO Should we call interface block members struct members?
+                while (b.getTokenType() == PREPROCESSOR_BEGIN) {
+                    parsePreprocessor();
+                }
```

Comment:
Do you have any example code which does not parse correctly without this?

Reply:
resources/environment/builtin.tese - if you check out the PSI tree, you'll notice the ifs being parsed as part of the previous lines
It doesn't pose that much issue now tho.